integration-tests/tests/src/tests/sync/mixed.ts

```diff
@@ -224,12 +239,79 @@ function describeTypes(flexibleSync: boolean) {
       result.value = result;
       return result;
     },
-    testValue: (value: MixedClass) => recursiveObjectId.equals(value._id),
-    flexibleSync,
+    valueTester: (value: MixedClass) => {
+      expect(recursiveObjectId.equals(value._id)).equals(true); //TODO I should be able to put this into the default tester
+    },
+    useFlexibleSync: flexibleSync,
   });
+
+  if (flexibleSync) {
+    describe("collections in mixed", () => {
+      const data = new Uint8Array([0xd8, 0x21, 0xd6, 0xe8, 0x00, 0x57, 0xbc, 0xb2, 0x6a, 0x15]);
+
+      const mixedList: Mixed[] = [
+        null,
+        true,
+        1,
+        5.0,
+        "string",
+        Realm.BSON.Decimal128.fromString("1234.5678"),
+        new Realm.BSON.ObjectId("609afc1290a3c1818f04635e"),
+        new Realm.BSON.UUID("9476a497-60ef-4439-bc8a-52b8ad0d4875"),
+        new Date(1620768552979),
+        data.buffer,
+      ];
+
+      const mixedDict = {
+        null: null,
+        bool: true,
+        int: 1,
+        float: 5.0,
+        string: "stringVal",
+        decimal: Realm.BSON.Decimal128.fromString("1234.5678"),
+        objectId: new Realm.BSON.ObjectId("609afc1290a3c1818f04635e"),
+        uuid: new Realm.BSON.UUID("9476a497-60ef-4439-bc8a-52b8ad0d4875"),
+        date: new Date(1620768552979),
+        data: data.buffer,
+      };
+
+      const nestedMixedList: Mixed[] = [...mixedList, mixedList, mixedDict];
+
+      const nextedMixedDict = {
+        ...mixedDict,
+        innerList: mixedList,
+        innerDict: mixedDict,
+      };
+
+      describeRoundtrip({
+        typeName: "list",
+        value: mixedList,
+        useFlexibleSync: true,
+      });
+
+      describeRoundtrip({
+        typeName: "nested list",
+        value: nestedMixedList,
+        useFlexibleSync: true,
+      });
+
+      describeRoundtrip({
+        typeName: "dictionary",
+        value: mixedDict,
+        useFlexibleSync: true,
+      });
+
+      describeRoundtrip({
+        typeName: "nested dictionary",
+        value: nextedMixedDict,
+        useFlexibleSync: true,
+      });
+    });
+  }
 }
 
-describe("mixed", () => {
+describe("mixed synced", () => {
+  //TODO Should we keep this around?
```

Comment:
As long as we support PBS, I think we should keep it

Reply:
Agreed